app/controllers/checkout_controller.rb

```diff
@@ -22,6 +22,12 @@ def update
       check_order_for_phantom_fees
       fire_event('spree.checkout.update')
       while @order.state != "complete"
+        if @order.order_cycle.present? && !recheck_stock_availability
+          flash[:error] = t('checkout.items_removed_from_oc')
+          raise_insufficient_quantity
+          return
+        end
+
```

Comment:
Shouldn't we being check this kind of stuff at model level instead of directly in the controller? In that way you ensure that we apply this rule every in every possible scenario, not just in this controller.